### PR TITLE
Disable tracer if it fails to start properly

### PR DESCRIPTION
**Pull request by richafrank**
Also fixed a typo.

Exceptions raised from `_init` meant that the singleton was left in a bad state.

### CODE REVIEW

File: qdb/tracer.py [hunk 1]
```diff
@@ -145,16 +152,17 @@ def _init(self, config=None, merge=False, **kwargs):
             self.log_handler.push_application()
 
         self.bound_cmd_manager = config.cmd_manager or TerminalCommandManager()
+        self.redirect_output = (
```
Comment (llllllllll) [inline qdb/tracer.py]: why did we move this away from the other output stuff?

File: qdb/comm.py [hunk 2]
```diff
@@ -366,7 +366,8 @@ def user_stop(self):
         """
         if self.reader and self.reader.is_alive():
             self.reader.terminate()
-        self.socket.close()
+        if self.socket is not None:
```
Comment (llllllllll) [inline qdb/comm.py]: with the changes to the `__new__`, I don't think we need these extra None checks. I am not sure we can hit these conditions anymore.

File: .travis.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 language: python
 python:
   - "2.7"
-  - "3.4"
+  - "3.5"
```
Comment (llllllllll) [inline .travis.yml]: is there a reason to make this 3.5 instead of 3.6 or 3.7?

Comment (richafrank) [reply .travis.yml]: I was keeping it up to date with zipline, but doesn't really have to be.

File: qdb/tracer.py (hunk 1 above)
Comment (richafrank) [reply qdb/tracer.py]: This was again accounting for missing attributes when exceptions were raised in `_init`. I'll look into undoing that, and leaving the singleton change.

File: .travis.yml (hunk 3 above)
Comment (richafrank) [reply .travis.yml]: Bumped it to 3.6 (travis didn't have 3.7)

**Pull request merged**